models/balance.js

```diff
@@ -1,52 +1,111 @@
-      WHERE
-        id = $1
-      LIMIT
-        1
-    ;`,
-    values: [operationId],
-  };
+async function findAllByOriginatorId(originatorId, options) {
+  const userTabcoins = await findAll({ balanceType: 'user:tabcoin', originatorId });
+  const userTabcash = await findAll({ balanceType: 'user:tabcash', originatorId });
+  const contentTabcoins = await findAll({ balanceType: 'content:tabcoin', originatorId });
+
+  return [...userTabcoins, ...userTabcash, ...contentTabcoins];
+
+  async function findAll({ balanceType, originatorId }) {
+    const tableNameMap = {
+      'user:tabcoin': 'user_tabcoin_operations',
+      'user:tabcash': 'user_tabcash_operations',
+      'content:tabcoin': 'content_tabcoin_operations',
+    };
+
+    const query = {
+      text: `
+        SELECT
+          *
+        FROM
+          ${tableNameMap[balanceType]}
+        WHERE
+          originator_id = $1
+      ;`,
+      values: [originatorId],
+    };
+
+    const results = await database.query(query, options);
+    return results.rows.map((balance) => mapBalanceOut(balance.balance_type ?? balanceType, balance));
+  }
+}
 
-  const results = await database.query(query, options);
-  return results.rows[0];
+function mapBalanceOut(balanceType, balance) {
+  balance.balance_type = balanceType.startsWith('user:') ? balanceType : `content:tabcoin:${balanceType}`;
+  return balance;
 }
 
-async function create({ balanceType, recipientId, amount, originatorType, originatorId }, options = {}) {
-  const query = {
-    text: `
-      INSERT INTO balance_operations
-        (balance_type, recipient_id, amount, originator_type, originator_id)
-      VALUES
-        ($1, $2, $3, $4, $5)
-      RETURNING * ;
-      `,
-    values: [balanceType, recipientId, amount, originatorType, originatorId],
-  };
+async function create(data, options) {
+  const create = data.balanceType.startsWith('user:') ? createUserBalance : createContentTabcoin;
+  return create(data, options);
 
-  const results = await database.query(query, options);
-  return results.rows[0];
+  async function createUserBalance(data, options) {
+    const table = data.balanceType === 'user:tabcoin' ? 'user_tabcoin_operations' : 'user_tabcash_operations';
+
+    const query = {
+      text: `
+        INSERT INTO ${table}
+          (recipient_id, amount, originator_type, originator_id)
+        VALUES
+          ($1, $2, $3, $4)
+        RETURNING *
+      ;`,
+      values: [data.recipientId, data.amount, data.originatorType, data.originatorId],
+    };
+
+    const results = await database.query(query, options);
+    const createdRow = results.rows[0];
+    return mapBalanceOut(data.balanceType, createdRow);
+  }
+
+  async function createContentTabcoin(data, options) {
+    const balanceTypeMap = {
+      'content:tabcoin:credit': 'credit',
+      'content:tabcoin:debit': 'debit',
+      'content:tabcoin:initial': 'initial',
+      'content:tabcoin': data.amount > 0 ? 'credit' : 'debit',
```

Comment:
Essa linha `'content:tabcoin': data.amount > 0 ? 'credit' : 'debit',` me fez entender o que você tinha comentado sobre alterar o `orchestrator`.

Isso só era necessário porque no  #1624 eu acabei esquecendo de ajustar os testes que chamavam `orchestrator.createBalance` passando `balanceType: 'content:tabcoin'`.

Na versão que eu sugeri agora, essa linha não é necessária, e na versão anterior, podia passar qualquer coisa, como neste exemplo:

```suggestion
      'content:tabcoin': 'qualquer_coisa_aqui_fazia_os_testes_passarem',
```

Pois os testes citados só estão usando o saldo total de TabCoins, então o valor de `balanceType` não importa.

Vou corrigir isso em outro PR, pois são coisas independentes.